libdevcore/JSON.h

```diff
@@ -23,22 +23,97 @@
 #pragma once
 
 #include <json/json.h>
+#include <sstream>
+#include <memory>
 
 namespace dev
 {
 
+namespace json
+{
+
+/// StreamWriterBuilder that can be constructed with specific settings
+class StreamWriterBuilder : public Json::StreamWriterBuilder {
+public:
+	explicit StreamWriterBuilder(const std::map<std::string, std::string> &_settings)
+	{
+		for (auto& iter : _settings)
+		{
+			this->settings_[iter.first] = iter.second;
+		}
+	}
+};
+
+/// CharReaderBuilder with strict-mode settings
+class StrictModeCharReaderBuilder : public Json::CharReaderBuilder {
+public:
+	StrictModeCharReaderBuilder()
+	{
+		Json::CharReaderBuilder::strictMode(&this->settings_);
+	}
+};
+
+/// Serialise the JSON object (@a _input) with specific builder (@a _builder)
+/// \param _input JSON input string
+/// \param _builder StreamWriterBuilder that is used to create new Json::StreamWriter
+/// \return serialized json object
+inline std::string print(Json::Value const &_input, Json::StreamWriterBuilder &_builder) {
+	std::stringstream stream;
+	std::unique_ptr<Json::StreamWriter> writer(_builder.newStreamWriter());
+	writer->write(_input, &stream);
+	return stream.str();
+}
+
+/// Parse a JSON string (@a _input) with specified builder (@ _builder) and writes resulting JSON object to (@a _json)
+/// \param _builder CharReaderBuilder that is used to create new Json::CharReaders
+/// \param _input JSON input string
+/// \param _json [out] resulting JSON object
+/// \param _errs [out] Formatted error messages
+/// \return \c true if the document was successfully parsed, \c false if an error occurred.
+inline bool parse(Json::CharReaderBuilder& _builder, const std::string &_input, Json::Value *_json, std::string *_errs)
```

Comment:
Instead of pointers can we use referneces?

Reply:
I also thought about this. The good thing with pointers are, that the direction of read/write is more visible - but thats mainly somehow just a matter of taste. Of course this would only make sense, if it's not only used in `libdevcore/JSON.h`.

Reply:
Our general guideline is to only use a pointer if the pointer can be nullptr and references otherwise.

Reply:
Ok. I will just change `_json` to be a reference.